Objects/dictobject.c

```diff
@@ -3929,6 +3929,33 @@ static PyMethodDef mapp_methods[] = {
     {NULL,              NULL}   /* sentinel */
 };
 
+
+// return 1 if the module `key` in `mp` is not loaded (a lazy key)
+// return 0 if the module `key` in `mp` is loaded
+// return -1 if existing an error
+int
+PyDict_IsLazyImport(PyObject *mp, PyObject *key)
+{
+    PyObject *value = _PyDict_GetItemKeepLazy(mp, key);
+
+    // error
+    if (value == NULL) {
+        PyErr_SetString(PyExc_KeyError,
+                        "cannot find this key "
+                        "in the specified dictionary");
+        return -1;
```

Comment:
as discussed in chat - either set the exception in the caller instead of here, or change this function to return `PyObject *`